### DISCUSSION TITLE
thumbfast在5月6日的mpv中出错了

### DISCUSSION THREAD

**Discussion by redomCL** [dyphire/mpv-config#86] [Q&A] [answered]
error:can't create mpv subprocess

**Comment by dyphire** [comment]
今天不才 5.5？你穿越了？
顺便应该是 mpv 上游添加的触控支持引入的错误，关注上游 issue：https://github.com/po5/thumbfast/issues/122

**Reply by redomCL** [reply]
打错了  是5月4号的

**Comment by dyphire** [comment]
由 https://github.com/mpv-player/mpv/commit/c4b6d0da3865d49c5c3df78902f6c4cc51cbcdbf 提交修复，今天的构建就会正常工作（尚未构建发布

**Reply by zhongfly** [reply, marked as answer]
不是mpv的问题，由https://github.com/FFmpeg/FFmpeg/commit/bb8044581366fe286e16b14515d873979133dbda 引起，然后被 https://github.com/FFmpeg/FFmpeg/commit/6a5b021e353655ddeb1a29f50e519f562dcae5a7 修复。
具体见https://github.com/mpv-player/mpv/issues/14047
https://github.com/zhongfly/mpv-winbuild/releases/tag/2024-05-04-637bc69 包含了ffmpeg修复补丁，不包含你说的 https://github.com/mpv-player/mpv/commit/c4b6d0da3865d49c5c3df78902f6c4cc51cbcdbf ，已经解决了这个问题，足以说明。

**Reply by dyphire** [reply]
确实如此

**Reply by zhongfly** [reply]
因为不想单开一个讨论，所以就直接在这里说了：
mpv现在添加了媒体控件的支持，并且默认启用，这会导致使用thumfast脚本后，会同时新建2个播放的显示，其中一个不能←→切换的就是脚本进程创建的。更奇怪的是，在另一个播放显示里← →切换的话，mpv窗口会冻结卡死。

所以必需在thumfast脚本里设置--media-controls=no禁用。至少我测试禁用了脚本的控件后，切换不会卡死了

包含媒体控件支持的mpv: https://github.com/zhongfly/mpv-winbuild/actions/runs/10153830186

**Reply by dyphire** [reply]
为什么会出现这种问题？是 thumfast 在子进程调用了 mpv 创建缩略图的工作方式导致的？

如果是这样，我感觉这是一个 mpv 问题，因为所有不使用 ffmpeg 而通过子进程调用 mpv 来剪切视频创建截图之类的第三方脚本应该都会触发这个问题

像kikoplay 这种调用多个 libmpv 的呢？它的进度条预览也是通过调用一个独立的 libmpv 实现的

**Reply by zhongfly** [reply]
> 为什么会出现这种问题？是 thumfast 在子进程调用了 mpv 创建缩略图的工作方式导致的？
> 
> 如果是这样，我感觉这是一个 mpv 问题，因为所有不使用 ffmpeg 而通过子进程调用 mpv 来剪切视频创建截图之类的第三方脚本应该都会触发这个问题
> 
> 像kikoplay 这种调用多个 libmpv 的呢？它的进度条预览也是通过调用一个独立的 libmpv 实现的

只要用mpv.exe播放就会有，我觉得这也是正常的。他们调用mpv的方法，和直接在cmd里运行mpv没区别。

libmpv不会的，选项的默认设置是只有mpv播放器才会启用控件，libmpv默认是不启用的。

另外，我前面说的控件会导致卡死反而复现不出来了，应该只是意外。

**Reply by dyphire** [reply]
media-controls=yes一样会为libmpv启用媒体控件

**Reply by zhongfly** [reply]
> media-controls=yes一样会为libmpv启用媒体控件

但默认不是yes，除非手动设置，对libmpv没影响的。

以后，相关的脚本都需要注意禁用这个选项罢了

**Reply by zhongfly** [reply]
哦，刚想到对于没有这个选项的版本来说，设置不存在的选项mpv会报错退出的，不能忽略错误
所以还得做版本判断，那是麻烦一些了

**Reply by dyphire** [reply]
使用你提供的编译版在 win11 上测试我没有复现你说的同时新建 2 个播放的显示，控件也正常。你是在 win10 上进行测试的？

Edit: 尽管无法复现，依然在新的提交 https://github.com/dyphire/mpv-config/commit/c8ecdc 中禁用，增强兼容性

**Reply by zhongfly** [reply]
我就是在win11上测试的，你只打开播放是不会触发thumbfast新建线程的，需要在进度条上触发一次thumbfast缩略图预览后才会新增一个，smtc右上角有一个>，点击可以切换显示
![image](https://github.com/user-attachments/assets/6aa20837-957b-4dc8-8358-3ac8b9cb8158)

**Reply by zhongfly** [reply]
建议在有多个文件/有播放列表 能切换视频的地方测试，不然2个控件比较难看出区别
![PixPin_2024-07-31_05-35-51](https://github.com/user-attachments/assets/70da7760-1b04-4c2e-b5af-f06c77985fe9)

**Reply by dyphire** [reply]
我其实有拖进度条预览和点击媒体控件的右上角切换，你一说我意识到确实有变化，不过控件按钮没变灰只是章节状态变了。不管了，反正已经修复了
